core/src/main/java/org/keycloak/representations/account/ClientRepresentation.java

```diff
@@ -1,11 +1,20 @@
 package org.keycloak.representations.account;
 
+import java.util.List;
+
 /**
  * Created by st on 29/03/17.
  */
 public class ClientRepresentation {
     private String clientId;
     private String clientName;
+    private String description;
+    private boolean internal;
```

Comment:
What is "internal"? Is it true for clients that don't require consent, and false for clients that do require consents? I don't think that's clear and could be easily mistaken. I would suggest userConsentRequired: true|false

Reply:
Perhaps combine with suggestion of consents object, so would be something like:

Third-party:
```
"consents": {
  userConsentRequired: true,
  grantedOnDate: date,
  updatedOnDate: date,
  scopes: [...]
}
```

Internal:
```
"consents": {
  userConsentRequired: false,
  scopes: [...]
}

Reply:
> Perhaps combine with suggestion of consents object, so would be something like:

@stianst What if a third-party client doesn't require consent? This way it'd be marked as "internal".

If I understand the wireframes correctly (where the "internal" status was clearly visible), "internal" is any client that's generated by Keycloak (Account Console, Admin Console, Admin CLI, ...).

Perhaps should we add a switch to the Admin Console to mark any client as internal, so admins could decide what "internal" really means?

Reply:
Internal as mentioned in the wire-frames are clients that do not require consent. So internal to the organization, not internal clients within Keycloak itself.

Reply:
Strictly speaking we are talking about clients that require consent and those that don't. End-user may not quite understand what consent means, which is why we had the slightly more higher level description of internal (to the organization) and third-party clients instead.

Reply:
@stianst Ok, makes sense, thanks for the clarification.